giskard_vision/landmark_detection/tests/base.py

```diff
@@ -301,9 +314,8 @@ def run(
         ground_truth_ref = dataloader_ref.all_marks
         metric_value_ref = self.metric.get(prediction_result_ref, ground_truth_ref)
 
-        if self.metric_for_examples is not None:
-            metrics_examples = self.metric_for_examples.get(prediction_result, ground_truth)
-            indexes = sorted(range(len(metrics_examples)), key=metrics_examples.__getitem__)[::-1]
+        if hasattr(self.metric, "rank_data"):
```

Comment:
careful: you did this couple of times in the code. if `rank_data` is defined as abstract method in the base class (not that it should be), you'll always have it as attr

Reply:
make `rank_data` return `None`, all this if else, will disappear